src/tree.h

```diff
@@ -12,17 +12,20 @@
 #include "odb.h"
 #include "vector.h"
 #include "strmap.h"
+#include "pool.h"
 
 struct git_tree_entry {
 	uint16_t attr;
 	git_oid oid;
+	bool pooled;
```

Comment:
This mis-aligns the struct packing, which is not neat for pool allocations. I think we could very easily change `filename_len` to `uint16_t` and get a couple extra bytes to play with.